### PR TITLE
[Intents] Add support for xcode 15.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by haritha-mohan — APPROVED** [review]

File: src/intents.cs [hunk 1]
```diff
@@ -4379,9 +4403,40 @@ interface INMessage : NSCopying, NSSecureCoding {
...
+
+		[NullAllowed]
+		[Watch (10, 0), NoMac, iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("attachmentFiles", ArgumentSemantic.Copy)]
+		INFile [] AttachmentFiles { get; }
+
+		[NullAllowed]
+		[Watch (10, 0), NoMac, iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("numberOfAttachments", ArgumentSemantic.Copy)]
+		NSNumber NumberOfAttachments { get; }
```
Comment (rolfbjarne) [inline src/intents.cs:4433]: BindAs?

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: src/intents.cs [hunk 2]
```diff
@@ -4379,9 +4403,40 @@ interface INMessage : NSCopying, NSSecureCoding {
...
+		[Watch (10, 0), NoMac, iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("initWithIdentifier:conversationIdentifier:content:dateSent:sender:recipients:groupName:serviceName:messageType:numberOfAttachments:")]
+		NativeHandle Constructor (string identifier, [NullAllowed] string conversationIdentifier, [NullAllowed] string content, [NullAllowed] NSDate dateSent, [NullAllowed] INPerson sender, [NullAllowed] INPerson [] recipients, [NullAllowed] INSpeakableString groupName, [NullAllowed] string serviceName, INMessageType messageType, [NullAllowed] NSNumber numberOfAttachments);
+
+		[NullAllowed]
+		[Watch (10, 0), NoMac, iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("attachmentFiles", ArgumentSemantic.Copy)]
+		INFile [] AttachmentFiles { get; }
+
+		[NullAllowed]
```
Comment (mandel-macaque) [inline src/intents.cs:4430]:
```suggestion
		[NullAllowed, 	BindAs (typeof (int))]
```

File: src/intents.cs [hunk 3]
```diff
@@ -14998,4 +15087,102 @@ interface INHangUpCallIntentResponse {
...
+	interface INEditMessageIntentHandling {
+		[Abstract]
+		[Export ("handleEditMessage:completion:")]
+		void HandleEditMessage (INEditMessageIntent intent, Action<INEditMessageIntentResponse> completion);
+
+		[Export ("confirmEditMessage:completion:")]
+		void ConfirmEditMessage (INEditMessageIntent intent, Action<INEditMessageIntentResponse> completion);
+
+		[Export ("resolveEditedContentForEditMessage:withCompletion:")]
+		void ResolveEditedContent (INEditMessageIntent intent, Action<INStringResolutionResult> completion);
```
Comment (tj-devel709) [inline src/intents.cs:15174]: Should these be async?

File: src/intents.cs [hunk 4]
```diff
@@ -14998,4 +15087,102 @@ interface INHangUpCallIntentResponse {
...
+
+	[Watch (10, 0), NoTV, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
+	interface INUnsendMessagesIntentHandling {
+		[Abstract]
+		[Export ("handleUnsendMessages:completion:")]
+		void HandleUnsendMessages (INUnsendMessagesIntent intent, Action<INUnsendMessagesIntentResponse> completion);
+
+		[Export ("confirmUnsendMessages:completion:")]
+		void ConfirmUnsendMessages (INUnsendMessagesIntent intent, Action<INUnsendMessagesIntentResponse> completion);
```
Comment (tj-devel709) [inline src/intents.cs:15185]: Should these be async as well?

File: src/intents.cs (hunk 3 above)
Comment (dalexsoto) [reply src/intents.cs:15174]: Ah I merged and your comments popped up. Async can't be applied in protocols

File: src/intents.cs (hunk 4 above)
Comment (dalexsoto) [reply src/intents.cs:15185]: Same, Async can't go in protocols

**Pull request merged**